### PR TITLE
Edwinn branch

### PR DESCRIPTION
**Pull request by edwinnsilva**

### CODE REVIEW

File: src/main/java/com/javatechie/springboot/crud/example/entity/Product.java [hunk 1]
```diff
@@ -22,5 +23,48 @@
     private int id;
     private String name;
     private int quantity;
+    private int status;
     private double price;
-}
+    private int categoria;
+
+    public static Product ProducttoDto(ProductDto productDto) throws Exception {
```
Comment (deiverrovira) [inline src/main/java/com/javatechie/springboot/crud/example/entity/Product.java]: Please change this method name to something meaningful, also please take into account that methods names should start with lowercase and usually the name should be a verb describing what the method does

Comment (edwinnsilva) [reply src/main/java/com/javatechie/springboot/crud/example/entity/Product.java:30]: The correct name for this method could  be "mapProductDTotoProduct" ?

Comment (edwinnsilva) [reply src/main/java/com/javatechie/springboot/crud/example/entity/Product.java]: The correct name for this method can be "mapProductDtoToProduct" ?

Comment (deiverrovira) [reply src/main/java/com/javatechie/springboot/crud/example/entity/Product.java]: yes it's perfect

File: src/main/java/com/javatechie/springboot/crud/example/entity/Product.java [hunk 2]
```diff
@@ -22,5 +23,48 @@
     private int id;
     private String name;
     private int quantity;
+    private int status;
     private double price;
-}
+    private int categoria;
+
+    public static Product mapProductDtoToProduct(ProductDto productDto) throws Exception {
```
Comment (deiverrovira) [inline src/main/java/com/javatechie/springboot/crud/example/entity/Product.java]: **Product** java class should be a POJO (representation of only fields with getters/setter). please move this method to another class, for example ProductTransformer,java

File: src/main/java/com/javatechie/springboot/crud/example/service/ProductService.java [hunk 3]
```diff
@@ -35,15 +33,23 @@ public Product getProductByName(String name){
...
+    public Product updateProduct(Product product) {
         Product existingProduct=repository.findById(product.getId()).orElse(null);
         existingProduct.setName(product.getName());
         existingProduct.setQuantity(product.getQuantity());
+        existingProduct.setStatus(product.getStatus());
         existingProduct.setPrice(product.getPrice());
         return repository.save(existingProduct);
 
     }
+
```
Comment (deiverrovira) [inline src/main/java/com/javatechie/springboot/crud/example/service/ProductService.java]: Please do not add empty lines

File: src/main/java/com/javatechie/springboot/crud/example/controller/ProductController.java [hunk 4]
```diff
@@ -1,25 +1,32 @@
...
 public class ProductController {
 
  @Autowired
  private ProductService service;
 
  @PostMapping("/addProduct")
- public Product addProduct(@RequestBody Product product){
-    return service.saveProduct(product);
+ public Product addProduct(@RequestBody ProductDto productDto) throws Exception {
+    return service.saveProduct(mapProductDtoToProduct(productDto));
```
Comment (deiverrovira) [inline src/main/java/com/javatechie/springboot/crud/example/controller/ProductController.java:22]: Well done, sometimes its better to do not import the static method, instead you'll see something like  this:
    return service.saveProduct(**ProductTransformer.**mapProductDtoToProduct(productDto));

**Review by deiverrovira — APPROVED** [review]
Looks good, Thank you

**Pull request merged**